File: javav2/example_code/codedeploy/src/main/java/com/example/deploy/GetDeployment.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.deploy;

// snippet-start:[codedeploy.java2._get_deployment.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.codedeploy.CodeDeployClient;
import software.amazon.awssdk.services.codedeploy.model.CodeDeployException;
import software.amazon.awssdk.services.codedeploy.model.GetDeploymentRequest;
import software.amazon.awssdk.services.codedeploy.model.GetDeploymentResponse;
// snippet-end:[codedeploy.java2._get_deployment.import]

/**
 * Before running this Java V2 code example, set up your development
 * environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class GetDeployment {

    public static void main(String[] args) {

        final String usage = """

                Usage:
                    <deploymentId>\s

                Where:
                    deploymentId - The id of the deployment.\s
                """;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String deploymentId = args[0];
        Region region = Region.US_EAST_1;
        CodeDeployClient deployClient = CodeDeployClient.builder()
                .region(region)
                .build();

        getSpecificDeployment(deployClient, deploymentId);
        deployClient.close();
    }

    // snippet-start:[codedeploy.java2._get_deployment.main]
    public static void getSpecificDeployment(CodeDeployClient deployClient, String deploymentId) {

        try {
            GetDeploymentRequest deploymentRequest = GetDeploymentRequest.builder()
                    .deploymentId(deploymentId)
                    .build();

            GetDeploymentResponse response = deployClient.getDeployment(deploymentRequest);
            System.out.println("The application associated with this deployment is "
                    + response.deploymentInfo().applicationName());

        } catch (CodeDeployException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
    // snippet-end:[codedeploy.java2._get_deployment.main]
}

File: .kotlin_alpha/services/codedeploy/src/main/kotlin/com/kotlin/deploy/GetDeployment.kt
//snippet-sourcedescription:[GetDeployment.kt demonstrates how to get information about a deployment.]
//snippet-keyword:[AWS SDK for Kotlin
//snippet-keyword:[Code Sample]
//snippet-keyword:[AWS CodeDeploy]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[10/28/2021]
//snippet-sourceauthor:[scmacdon AWS]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.deploy

// snippet-start:[codedeploy.kotlin._get_deployment.import]
import aws.sdk.kotlin.services.codedeploy.CodeDeployClient
import aws.sdk.kotlin.services.codedeploy.model.CodeDeployException
import aws.sdk.kotlin.services.codedeploy.model.GetDeploymentRequest
import kotlin.system.exitProcess
// snippet-end:[codedeploy.kotlin._get_deployment.import]

suspend fun main(args:Array<String>) {

    val usage = """
    Usage:
        <deploymentId> 

    Where:
       deploymentId - the id of the deployment. 
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(1)
    }

    val deploymentId = args[0]
    val codeDeployClient = CodeDeployClient{region ="us-east-1"}
    getSpecificDeployment(codeDeployClient, deploymentId)
    codeDeployClient.close()
}

// snippet-start:[codedeploy.kotlin._get_deployment.main]
suspend fun getSpecificDeployment(deployClient: CodeDeployClient, deploymentIdVal: String?) {
    try {
        val deploymentRequest = GetDeploymentRequest {
            deploymentId= deploymentIdVal
        }

        val response = deployClient.getDeployment(deploymentRequest)
        println("The application associated with this deployment is ${response.deploymentInfo?.applicationName}")

    } catch (e: CodeDeployException) {
        System.err.println(e.message)
        exitProcess(0)
    }
}
// snippet-end:[codedeploy.kotlin._get_deployment.main]